api/src/test/java/org/openmrs/api/context/ContextTest.java

```diff
@@ -238,4 +254,111 @@ public void becomeUser_shouldChangeLocaleWhenBecomeAnotherUser() {
 		
 		Context.logout();
 	}
+
+	/**
+	 * @see Context#evictSingleEntity(SessionFactory, Class, String)
+	 */
+	@Test
+	public void evictSingleEntity_shouldClearSingleEntityFromCaches(){
+		//SessionFactory sf = (SessionFactory) applicationContext.getBean("sessionFactory");
+		PersonName name = Context.getPersonService().getPersonName(PERSON_NAME_ID_2);
+		//Load the person so that the names are also stored  in person names collection region
+		Context.getPersonService().getPerson(name.getPerson().getPersonId());
+
+		//Let's have the name in a query cache
+		sf.getCurrentSession().createCriteria(PERSON_NAME_CLASS)
+			.add(Restrictions.eq("personNameId", PERSON_NAME_ID_2))
+			.setCacheable(true)
+			.setCacheRegion(QUERY_REGION)
+			.list();
+		assertTrue(sf.getCache().containsEntity(PERSON_NAME_CLASS, PERSON_NAME_ID_2));
+		assertNotNull(sf.getStatistics().getSecondLevelCacheStatistics(Person.class.getName() + ".names")
+			.getEntries().get(name.getPerson().getPersonId()));
+		assertEquals(1, sf.getStatistics().getCacheRegionStatistics(QUERY_REGION).getRegionName());
```

Comment:
This is also returning a NPE because at the level, `getCacheRegionStatistics(QUERY_REGION)`

cc @wluyima @ibacher